src/systems/physics/Physics.cc

```diff
@@ -4165,6 +4179,66 @@ void PhysicsPrivate::UpdateCollisions(EntityComponentManager &_ecm)
       });
 }
 
+//////////////////////////////////////////////////
+void PhysicsPrivate::UpdateRayIntersections(EntityComponentManager &_ecm)
+{
+  GZ_PROFILE("PhysicsPrivate::UpdateRayIntersections");
+  // Quit early if the MultiRayIntersections component hasn't been created.
+  // This means there are no systems that need contact information
+  if (!_ecm.HasComponentType(components::MultiRayIntersections::typeId))
+    return;
+
+  // Assume that there is only one world entity
+  Entity worldEntity = _ecm.EntityByComponents(components::World());
+
+  if (!this->entityWorldMap.HasEntity(worldEntity))
+  {
+    gzwarn << "Failed to find world [" << worldEntity << "]." << std::endl;
+    return;
+  }
+
+  auto worldRayIntersectionFeature =
+      this->entityWorldMap.EntityCast<RayIntersectionFeatureList>(worldEntity);
+
+  // Go through each entity that has a MultiRay and MultiRayIntersections
+  // components, trace the rays and set the MultiRayIntersections component
+  // value to the list of intersections that correspond to the ray entity
+  _ecm.Each<components::MultiRay,
+            components::MultiRayIntersections>(
```

Comment:
Does this mean the user has to create two components on the entity to use this feature? Would it be possible to combine them into one? Or we can have the physics engine be responsible for creating the result component (components::MultiRayIntersections).

Reply:
Good suggestion!

I unified these two components in the `RaycastData` one. Done in https://github.com/gazebosim/gz-sim/pull/2514/commits/16791b71e8626cfec5e778677c7080c05b71866a